### PR TITLE
feat: add terraform for azure functions

### PR DESCRIPTION
**Pull request by saulfrance**
#### 📲 What

Adds Terraform for Azure functions.

#### 🤔 Why

So the user can deploy the dotnet function apps.

#### 🛠 How

- Added a function module with a `create_function` condition.
- Removed all previous function logic from the service bus module.

#### ✅ Acceptance criteria Checklist

- [ ] Code peer reviewed?
- [ ] Documentation has been updated to reflect the changes?
- [ ] Passing all automated tests, including a successful deployment?
- [ ] Passing any exploratory testing?
- [ ] Rebased/merged with latest changes from development and re-tested?
- [ ] Meeting the Coding Standards?

### CODE REVIEW

File: deploy/azure/app/kube/main.tf [hunk 1]
```diff
@@ -15,7 +15,7 @@ module "default_label" {
 }
 
 module "app" {
-  source                               = "git::https://github.com/amido/stacks-terraform//azurerm/modules/azurerm-server-side-app?ref=v1.5.1"
+  source                               = "git::https://github.com/ensono/stacks-terraform//azurerm/modules/azurerm-server-side-app?ref=v1.5.1"
```
Comment (alexanderwjrussell) [inline deploy/azure/app/kube/main.tf:18]: Should this have a double / in the middle? 
`stacks-terraform//azurerm`

File: deploy/azure/app/function/constraints.tf [hunk 2]
```diff
@@ -0,0 +1,9 @@
+terraform {
+  required_version = ">= 1.1"
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = "~> 2.98"
```
Comment (sharonrussell) [inline deploy/azure/app/function/constraints.tf:6]: is this the latest version we can use?

File: deploy/azure/app/kube/main.tf (hunk 1 above)
Comment (saulfrance) [reply deploy/azure/app/kube/main.tf:18]: Yeah that is normal for Terraform modules. Everything before the `//` is the repo and everything after is the path to the file.

https://developer.hashicorp.com/terraform/language/modules/sources#modules-in-package-sub-directories

File: deploy/azure/app/servicebus/outputs.tf [hunk 3]
```diff
@@ -32,19 +32,3 @@ output "servicebus_subscription_filtered_id" {
   description = "Servicebus Subscription filtered ID"
```
Comment (alexanderwjrussell) [inline deploy/azure/app/servicebus/outputs.tf:32]:
```suggestion
  description = "ServiceBus Subscription filtered ID"
```

File: deploy/azure/app/function/constraints.tf (hunk 2 above)
Comment (saulfrance) [reply deploy/azure/app/function/constraints.tf:6]: Most likely not but I've keep it consistent with the rest of the Terraform in the repo. There are also a bunch of warnings produced when we run it. I'll create another story to upgrade providers and address warnings

**Review by alexanderwjrussell — APPROVED** [review]
Looks good to me. Just a couple of minor comments

**Review by sharonrussell — APPROVED** [review]

File: deploy/azure/app/function/constraints.tf
[the remark this answers is not in the document]
Comment (saulfrance) [reply deploy/azure/app/function/constraints.tf:1]: Updated

File: deploy/azure/app/kube/main.tf [hunk 4]
```diff
@@ -57,3 +50,17 @@ module "eventhub" {
...
+  function_name                = var.function_name
+  resource_group_name          = module.default_label.id
+  resource_group_location      = var.resource_group_location
+  cosmosdb_database_name       = module.app.cosmosdb_database_name
+  cosmosdb_collection_name     = var.cosmosdb_sql_container
+  cosmosdb_connection_string   = "AccountEndpoint=${module.app.cosmosdb_endpoint};AccountKey=${module.app.cosmosdb_primary_master_key};"
+  sb_topic_name                = module.servicebus[0].servicebus_topic_name
+  sb_subscription_name         = module.servicebus[0].servicebus_subscription_name
+  servicebus_connection_string = module.servicebus[0].servicebus_connectionstring
+}
```
[the remark this answers is not in the document]
Comment (saulfrance) [reply deploy/azure/app/kube/main.tf]: Added

File: deploy/azure/app/kube/vars.tf
[the remark this answers is not in the document]
Comment (saulfrance) [reply deploy/azure/app/kube/vars.tf:1]: Updated

File: deploy/azure/app/servicebus/vars.tf
[the remark this answers is not in the document]
Comment (saulfrance) [reply deploy/azure/app/servicebus/vars.tf:1]: Updated

**Pull request merged**